packages/davinci-client/src/lib/client.store.ts

```diff
@@ -24,7 +25,19 @@ import type { SingleValueCollector } from './collector.types.js';
 export async function davinci({ config }: { config: DaVinciConfig }) {
   const store = createClientStore();
 
-  store.dispatch(configSlice.actions.set(config));
+  if (!config.serverConfig.wellknown) {
+    throw Error('wellknown endpoint required as part of the `config.serverOptions` ');
+  }
+
+  const { data: openIdResponse } = await store.dispatch(
+    wellknownApi.endpoints.wellknown.initiate(config.serverConfig.wellknown),
+  );
+
+  if (!openIdResponse) {
+    throw Error('error fetching wellknown response');
+  }
+
+  store.dispatch(configSlice.actions.set({ ...config, endpoints: openIdResponse }));
```

Comment:
Pass in the entire wellknown response here, open to changing the key `endpoints` to something else, @cerebrl like `openIdConfiguration`?